pilot/pkg/model/push_context_test.go

```diff
@@ -859,6 +859,47 @@ func TestIsServiceVisible(t *testing.T) {
 			},
 			expect: false,
 		},
+		{
+			name:        "service visible to none",
+			pushContext: &PushContext{},
+			service: &Service{
+				Attributes: ServiceAttributes{
+					Namespace: "bar",
+					ExportTo: map[visibility.Instance]bool{
+						visibility.None: true,
+					},
+				},
+			},
+			expect: false,
+		},
+		{
+			name:        "service has both public visibility and none visibility",
+			pushContext: &PushContext{},
+			service: &Service{
+				Attributes: ServiceAttributes{
+					Namespace: "bar",
+					ExportTo: map[visibility.Instance]bool{
+						visibility.Public: true,
+						visibility.None:   true,
```

Comment:
I think this is not a right config, maybe we should prevent it ealier

Reply:
Current code supports it and the comments are very clear on what to do when this is configured. That is why I added these tests to ensure the order is not messed up in future. I do not know if we can prevent this/change this now for BC reasons.